controllers/agentaction_controller.go

```diff
@@ -165,16 +165,17 @@ func (r *AgentActionReconciler) applyJobToStatus(log logr.Logger, action *porter
 		setCondition(log, action, porterv1.ConditionStarted, "JobStarted")
 	}
 
+conditions:
 	for _, condition := range job.Status.Conditions {
 		switch condition.Type {
 		case batchv1.JobComplete:
 			action.Status.Phase = porterv1.PhaseSucceeded
 			setCondition(log, action, porterv1.ConditionComplete, "JobCompleted")
-			break
+			break conditions
```

Comment:
I'm unclear on the intent of the change and what we want the code to be doing. The way I'm reading this, when a condition on the job is complete/failed, we will call setCondition, and then repeat the entire for loop over again, re-evaluating the original condition that called the goto.

Is there a way we can do this without the goto statement (break to a label)? At the very least we need a lot more comments explaining what's intended and make sure we have a test to validate that this can't cause an infinite loop.

Reply:
I misunderstood the original code. I think the break statement is redundant and I just removed them